libsolidity/ast/AST.cpp

```diff
@@ -648,20 +649,21 @@ bool Literal::looksLikeAddress() const
 	if (!isHexNumber())
 		return false;
 
-	return abs(int(value().length()) - 42) <= 1;
+	int const length = (int) count_if(next(next(begin(value()))), end(value()), [](char c) { return isxdigit(c); });
+	return abs(length - 40) <= 1;
```

Comment:
Might be easier to do something like `length >= 39 || length <= 41` - it does not need the cast from size_t to int.